cmd/demo/node.go

```diff
@@ -258,14 +262,57 @@ func (n *node) listen() error {
 	if err != nil {
 		return errors.WithMessage(err, "could not start tcp listener")
 	}
-	go n.client.HandleChannelProposals(n)
+	db, err := leveldb.LoadDatabase(config.DatabasePath)
+	if err != nil {
+		return errors.WithMessage(err, "creating/loading database")
+	}
+	persister, err := keyvalue.NewPersistRestorer(db)
+	if err != nil {
+		return errors.WithMessage(err, "creating PersistRestorer")
+	}
+	n.client.EnablePersistence(persister)
+
+	go n.client.Handle(n, n)
+	n.client.OnNewChannel(func(ch *client.Channel) {
```

Comment:
With everything that is going on in this single function `listen` now, it should be renamed to something like `setup`. `listen` is misleading.